### ISSUE TITLE
Support the sidewalk tag on highways

### ISSUE THREAD

**Issue by smsm1** [openstreetmap/iD#1762]
Would it be possible for the sidewalk tag to be supported in the presets so encourage more people to enter the data?

**Comment by cliffordsnow** [comment]
It looks the the way now has a direction arrow. That should remove one of the roadblocks to implementing this feature.

**Comment by boothym** [comment]
Would be nice to have a field for sidewalk=both/left/right/no to make it easier to add them - there's plenty of other fields available on highways but no sidewalk tag! Ideally it would not appear on motorways (in the UK) and would only have those four values available.

@bhousel how would we go about adding this?

**Comment by bhousel** [comment]
One way would be to just copy everything the [cycleway field](https://github.com/openstreetmap/iD/blob/master/modules/ui/fields/cycleway.js) does and make a new `sidewalk` field.

However I'd prefer to include _on-street_ parking, cycle lanes, and sidewalks into the lane editing UI that @kepta is building (#3333).   This would make the existing cycleway tag redundant, and free up that UI space for the generalized lane editor.

**Comment by bhousel** [comment]
stale

**Comment by quincylvania** [comment]
A lot's changed since 2013! The general preference is now to map sidewalks independently from roads. I'm closing this on that basis, in line with #5659. Though of course mappers are of course free to tag sidewalks whichever way they wish.

Any further consideration should be as part of a combined lane editor (#387).

**Comment by boothym** [comment]
>The general preference is now to map sidewalks independently from roads. 

What is the evidence for this?

**Comment by quincylvania** [comment]
> What is the evidence for this?

@boothym Usage of the `highway=sidewalk` tag has [grown faster](https://taghistory.raifer.tech/#***/footway/sidewalk&***/sidewalk/) than that of `sidewalk=`, and has about caught up in [total uses](https://taginfo.openstreetmap.org/tags/footway=sidewalk). Higher-resolution aerial imagery is more widely available, making it much easier to trace sidewalks in urban areas. Separate ways enable granular mapping for accessible pedestrian routing, since you can easily tag surface, smoothness, curbs, tactile paving, etc. separately from the road. There's a [whole website](https://www.opensidewalks.com/) about it.

**Comment by Nakaner** [comment]
@quincylvania wrote:
> @boothym wrote:
> > What is the evidence for this?
> 
> @boothym Usage of the `highway=sidewalk` tag has [grown faster](https://taghistory.raifer.tech/#***/footway/sidewalk&***/sidewalk/) than that of `sidewalk=`, and has about caught up in [total uses](https://taginfo.openstreetmap.org/tags/footway=sidewalk). Higher-resolution aerial imagery is more widely available, making it much easier to trace sidewalks in urban areas. Separate ways enable granular mapping for accessible pedestrian routing, since you can easily tag surface, smoothness, curbs, tactile paving, etc. separately from the road. There's a [whole website](https://www.opensidewalks.com/) about it.

TL;DR The model (counting TagHistory/global Taginfo) you base your decision on is not suitable to support forming a well though out decision.

@quincylvania, the explaination for your decision not to support it lacks a couple of things:

You say that `footway=sidewalk` grows faster than `sidewalk=*`. First, TagHistory has not been updated for about one to two years. Things can change during that time. Second, deciding with numbers should happen with care (otherwise you can just fall back to simply following what someone wrote on the wiki [1]). Imports or a large organised mapping acitvity in one region can influence a tag. Sidewalk mapping as separate ways happened as part of organised activities in various occiasions, in Austria, Germany (at least twice), US etc. There is no broad consensus whether to map sidewalks as separate ways or just as `sidewalk=*` tag on the road. Both options have their advantages and disadvantages. Opinions how to map sidewalks differ between regions. German mappers [tend](https://forum.openstreetmap.org/viewtopic.php?id=66029) to be against separately mapped sidewalks but it's not a broad consensus for anything apart from "We agree that we disagree and we avoid conflicts between each other".

To back this with numbers: According to global Taginfo, we have 1,688,300 `footway=sidewalk` vs. 1,699,339 `sidewalk=*` (out of them 512,928 `sidewalk=no/separate). However, different regions have different mapping behaviour:

* North America: 905,615 `footway=sidewalk`, 661,568 `sidewalk=*` (383,920 `sidewalk=no/none/separate`)
* Europe: 498 994 `footway=sidewalk`, 874,341 `sidewalk=*` (240,917 `sidewalk=no/separate`)
* Germany: 41,924 `footway=sidewalk`, 383,398 `sidewalk=*` (117,063 `sidewalk=no/separate`)
* France: 35,035 `footway=sidewalk`, 54,624 `sidewalk=*` (9,264 `sidewalk=no/separate`)
* Czech Republic: 11,821 `footway=sidewalk`, 7,250 `sidewalk=*` (2,434 `sidewalk=no/separate`)

Regional numbers come from Geofabrik regional Taginfo service, all regions mentioned above as defined by their Geofabrik clipping polygons.

Even looking at the numbers of Taginfo, even if done for all countries individually, cannot give a complete picture. I have seen too much separately mapped sidewalks for the last years without `footway=sidewalk` and their main road without `sidewalk=separate`. Even those who are in favour of mapping sidewalks as a tag on the road, usually agree that sidewalks physically separated from the road (flowerbed, handrail etc.) should be mapped as a separate way. If you base your decision on counting the numbers this way only, you base your decision on a very imprecise model.

What is wrong with adding a simple field for `sidewalk` to most road presets as it happens with `cuisine` and other additional features? Supporting the long list of tags for the properties of sidewalks `surface:sideway=*, surface:sidewalk:right=*, width:sidewalk=*, …` is difficult if it should be user friendly but that's not what was requested here.

How should closing this issue be read as?

1. The maintainers of iD thinks that mapping sidewalks separate is the only valid way no matter what local communities in some regions think? How should they explain to newbies/other mappers that they should ignore what the editor believed being official says/supports?
2. The maintainers prefers to invest his time efficiently and does not want to support two concurring solutions. Closing the issue means nothing more than "not on his todo/wish list".
3. The maintainers prefers to invest his time efficiently and does not want to support two concurring solutions. Closing the issue means that it is unlikely that pull requests adding support for the other common way of mapping sidewalks are unlikely to be accepted.
4. It is technical impossible to make presets support both sidewalks mapped separately and as a property of the road.

iD has provoked [lots of conflicts](https://wiki.openstreetmap.org/wiki/ID/Controversial_Decisions) in the past. Their number and their lacking resolution lead to the you, Qunicy, and the OSMF board [work](https://lists.openstreetmap.org/pipermail/osmf-talk/2020-June/006865.html) on ways "toward resolution of controversies related to iD". They all relate to decisions on tagging which could be summarised as "failing to compromise". I would like to see the future of iD not being as full of conflicts as the era prior to you joining as maintainer. It is up to you to make it happen.

[1] For the sake of clearance: Anyone can write anything there and blindly following it can be wrong.

**Comment by Mashin6** [comment]
I agree that the creators of iD too often decide/or influence tagging without getting proper approval from OSM community through official channels (and in all seriousness this should change). However, in this case tagging as separate ways is officially approved scheme, while tagging on roads had never reached that status. I think that as a community we should strive to keep the tagging as consistent and uniform as possible. It is ridiculous that mapmakers and data users have to maintain two systems for sidewalk routing. This is something that OSM community have to fix for ourself and not only for this case.

**Comment by kymckay** [comment]
@Mashin6 I'm curious because this comes up a lot: what official channels do you speak of and who represents the OSM community? 🤔

I think all involved in developing iD would actually be relieved if there were official channels coordinating sensible and well thought out centralised tagging scheme development as it would make their lives much easier. To my knowledge, no such thing exists. Closest thing is the wiki tagging voting process which has it's own inherent problems and doesn't have enough engagement to truly represent everyone.

**Comment by quincylvania** [comment]
@Nakaner Closing this issue means only that I don't think we should add a `sidewalk` field to roads. iD has gone without this field for seven years, so the state of play remains unchanged.

When there are multiple valid ways to map something in OSM, we try to surface just one in iD in order to avoid confusing or overwhelming new mappers. People are free to use whatever pattern they or their communities prefer. Mappers that want a more flexible editor can use JOSM or fork iD.

**Comment by Mashin6** [comment]
@SilentSpike OSM is a community based project, which comes with advantages and disadvantages. The decision making is largely decentralized, without any official entity that would issue directives. Although I personally think that it is a time to establish at least some form of committee that would help to streamline things and make communication more efficient. 

For now the best place to engage in conversation is on [mailing list](https://lists.openstreetmap.org/listinfo/tagging), [IRC](https://irc.openstreetmap.org/), some form of help can be provided by [DWG group](https://wiki.openstreetmap.org/wiki/Data_working_group). There is also [US Slack channel](osmus.slack.com), but it is only US specific and I would treat is as such.

**Comment by amandasaurus** [comment]
> tagging as separate ways is officially approved scheme

Huh? OSM has no officially approved tagging schemes for anything.

> as a community we should strive to keep the tagging as consistent and uniform as possible

I 100% agree.

> centralised tagging scheme

I'm personally not in favour of that, and it would bring many disadvantages.

> There is also US Slack channel, but it is only US specific and I would treat is as such.

There's plenty of non-US channels & non-US activity on that.

**Comment by boothym** [comment]
@quincylvania Given that a length of road with a sidewalk on either side of the carriageway would be tagged only once with sidewalk=both, but twice with two footway=sidewalk ways, it's not surprising that footway=sidewalk has caught up!

Maybe it's the general preference to map sidewalks independently from roads in the USA (where separate sidewalks outnumber the sidewalk tag by ~200k) but there is more to the world than the USA. The website you link to seems to be US-centric as well. 

Plus it doesn't take long to find examples of separate sidewalks which have been mapped badly and don't have all the proper connections to the roads, meaning they are almost useless for proper navigation. And they don't always represent the reality on the ground, in countries where pedestrians can cross anywhere they like rather than at a marked crossing.

>Closing this issue means only that I don't think we should add a sidewalk field to roads. iD has gone without this field for seven years, so the state of play remains unchanged.

As you can see from the thread I was dissuaded four years by Bryan otherwise I'd have submitted a PR just copying the cycleway field...

**Comment by Mashin6** [comment]
@rory Separate sidewalks were approved by majority of voting members. Discussion about semantics of "official" are not pertinent in this case. Anyways, I think that Github is not a place for these type of discussions. If you want to talk more hit me up in Slack or PM on osm.org

**Comment by cliffordsnow** [comment]
On Wed, Jul 1, 2020 at 10:27 AM Michael Booth <notifications@github.com>
wrote:

> @quincylvania <https://github.com/quincylvania> Given that a length of
> road with a sidewalk on either side of the carriageway would be tagged only
> once with sidewalk=both, but twice with two footway=sidewalk ways, it's not
> surprising that footway=sidewalk has caught up!
>
> Maybe it's the general preference to map sidewalks independently from
> roads in the USA (where separate sidewalks outnumber the sidewalk tag by
> ~200k) but there is more to the world than the USA. The website you link to
> seems to be US-centric as well.
>
> Plus it doesn't take long to find examples of separate sidewalks which
> have been mapped badly and don't have all the proper connections to the
> roads, meaning they are almost useless for proper navigation. And they
> don't always represent the reality on the ground, in countries where
> pedestrians can cross anywhere they like rather than at a marked crossing.
>
> Closing this issue means only that I don't think we should add a sidewalk
> field to roads. iD has gone without this field for seven years, so the
> state of play remains unchanged.
>
> As you can see from the thread I was dissuaded four years by Bryan
> otherwise I'd have submitted a PR just copying the cycleway field...
>
>
> Michael,
I'm one of those mappers that switch from mapping sidewalks as attributes
of streets to separately mapped ways. The reason I switched was that it's
nearly impossible to route pedestrians using the attribute method.
Especially for pedestrians with limited mobility. The attribute method
basically just informs vehicle traffic that a sidewalk is alongside the
road. Mapping as separate ways allows the same level of detail that we
provide vehicles for roads. Just adding footway=sidewalk tells the
pedestrian that the way is beside a road.

I'm actually surprised that the US took the lead on this. We are such a car
centric country. As a lifetime walker, I've been hit by a car and can't
begin to count the number of times that a car didn't even see me. Just look
at Google pedestrian routing. It is terrible. We can do better. As for the
number of separate sidewalks, I am one of those that have mapped thousands
of km of sidewalks. With the help of others, my entire county is just about
complete.

To your point that separate sidewalks have some poor mapping. I'm sure I'm
one of those making mistakes although I do try to go back and review my
work. One of the problems is that there aren't many good pedestrian routing
websites. Another problem is that sidewalks are notorious for bad planning.
City planners around here seem to look at them not as a route, but
something that is required. I use a QGIS plugin, disconnect islands, to map
out all of the sidewalks that don't connect. Again, the problem goes back
to the US being a car centric society. Hopefully OSM can help improve that.

Best,
Clifford

-- 
@osm_washington
www.snowandsnow.us
OpenStreetMap: Maps with a human touch
